src/index.js

```diff
@@ -323,38 +363,40 @@ class KadDHT extends EventEmitter {
    * Search for a peer with the given ID.
    *
    * @param {PeerId} id
-   * @param {Object} options - findPeer options
-   * @param {number} options.timeout - how long the query should maximally run, in milliseconds (default: 60000)
+   * @param {Object} [options] - findPeer options
+   * @param {number} [options.timeout=60000] - how long the query should maximally run, in milliseconds (default: 60000)
    * @returns {Promise<{ id: PeerId, multiaddrs: Multiaddr[] }>}
    */
-  async findPeer (id, options = {}) { // eslint-disable-line require-await
+  async findPeer (id, options = { timeout: 60000 }) { // eslint-disable-line require-await
     return this.peerRouting.findPeer(id, options)
   }
 
   /**
    * Kademlia 'node lookup' operation.
+   *
    * @param {Uint8Array} key
    * @param {Object} [options]
-   * @param {boolean} [options.shallow] shallow query (default: false)
-   * @returns {AsyncIterable<{ id: PeerId, multiaddrs: Multiaddr[] }>}
```

Comment:
@achingbrain can we have this back?

Reply:
It's best to let tsc infer the return type if possible as it'll show you what the return type actually is, rather than what you think it is.

If the manually specified return type disagrees with the type tsc infers, it'll take the manually specified type which can introduce bugs when you get it wrong, so as a rule of thumb we should let tsc infer the return type unless it's impossible to express due to weird logic like [this sort of thing](https://github.com/IndigoUnited/js-err-code/blob/master/index.js#L58-L65).

Reply:
That said, we have overridden the return type for generator functions to say they return async iterables in other places (interface-datastore for example).  With hindsight I'm not 100% this was a great idea as it's not the actual return type and it prevents (or at least discourages) the caller from using the async generator API.

Reply:
ok, fair enough!